### ISSUE TITLE
appSecurity-3.0 slower than appSecurity-2.0 due to jaspic check.

### ISSUE THREAD

**Issue by jdmcclur** [OpenLiberty/open-liberty#15405]
I am noticing this extra time spent in an mp-4.0 app compared to a mp-3.3 app. (JaspiServiceImpl.isAnyProviderRegistered)

```
  Parent      0   0.11   2.26           7         147    J:com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImpl.performSecurityChecks(Ljavax/servlet/http/HttpServletRequest;Ljavax/servlet/http/HttpServletResponse;Ljavax/security/auth/Subject;Lcom/ibm/ws/webcontainer/security/WebSecurityContext;)V
    Self      0   0.11   2.26           7         147    J:com/ibm/ws/security/jaspi/JaspiServiceImpl.isAnyProviderRegistered(Lcom/ibm/ws/webcontainer/security/WebRequest;)Z
   Child      0   1.22   2.08          79         135    J:com/ibm/ws/security/javaeesec/BridgeBuilderImpl.buildBridgeIfNeeded(Ljava/lang/String;Ljavax/security/auth/message/config/AuthConfigFactory;)V
   Child      0   0.06   0.06           4           4    J:com/ibm/wsspi/kernel/service/utils/AtomicServiceReference.getService()Ljava/lang/Object;
   Child      0   0.00   0.02           0           1    J:com/ibm/ws/security/jaspi/JaspiServiceImpl.getAuthConfigFactory()Ljavax/security/auth/message/config/AuthConfigFactory;
```

It looks like mpJwt-1.2 pulls in appSecurity-3.0 (while mp3.3 used appSecurity-2.0), so now isJapsiEnabled is true here, which leads to the extra 2.26% time seen above:
https://github.com/OpenLiberty/open-liberty/blob/c2b0154dcec9728aea7bf3bb72b8c7d99f18b7d7/dev/com.ibm.ws.webcontainer.security/src/com/ibm/ws/webcontainer/security/WebAppSecurityCollaboratorImpl.java#L667

Is there anyway to disable jaspic since I am not using it, or is this something spec-wise we’re stuck with? I also wonder why it is doing the buildBridgeIfNeeded on every request,

**Comment by teddyjtorres** [comment]
We need to determine if we can back level to appSecurity-2.0.

It needs to call BridgeBuilderImpl.buildBridgeIfNeeded on every request since the internal JASPIC provider to be created is application specific. Once it has been determined that there is a provider for this application, isAnyProviderRegistered will return quickly instead of creating the provider.

**Comment by jdmcclur** [comment]
@teddyjtorres Any update on this?

**Comment by teddyjtorres** [comment]
We cannot use appSecurity-2.0. It does not work with jakarta, only appSecurity-3.0. The difference in perf is due to using appSecurity-3.0 and it cannot be modified.

**Comment by Emily-Jiang** [comment]
@teddyjtorres in appsecurity 3.0, is Jaspi always enabled? Do we need it to be enbled for mpJwt-1.2?

**Comment by jdmcclur** [comment]
Closed via PR #18399. Still some overhead, but it is minimal.
